rfcs/0145-doc-strings.md

```diff
@@ -0,0 +1,501 @@
+shepherd-team: (@DavHau; tbd.)
+shepherd-leader: (name to be appointed by RFC steering committee)
+related-issues: (will contain links to implementation PRs)
+---
+
+# Summary
+[summary]: #summary
+
+Propose a standard format specification for Doc-comments.
+
+This RFC includes two concerns that define a doc-comment:
+
+- Outer format rules to allow distinction between regular comments and doc-comments
+- Inner format rules that describe the required format of the content. 
+
+However, both concerns relate closely to each other; It makes sense and reduces bureaucracy to address that in a single RFC.
+
+# Motivation
+[motivation]: #motivation
+
+The following are the envisioned goals.
+
+- Be able to generate documentation from code for any nix expression.
+- Be able to distinguish between documentation-relevant comments and unrelated comments.
+- Make doc comments easy to write and read
+- Be able to parse and render doc comments nicely
+- Standardize a format for doc comments that further RFCs can extend
+
+This RFC is a significant change to existing documentation conventions.
+It allows distinguishing between regular and doc comments. Having distinction is essential because arbitrary code comments should not end up in generated documentation.
+
+> Hint: Generating static documentation is controvert topic in nixpkgs. We found that it is impossible to generate accurate documentation statically. A correct solution would involve the evaluation of expressions in some way.
+
+## Current State
+
+Currently, Nix does not have doc comments. Over time there evolved certain conventions, but they were never formally specified.
+Within this RFC, we want to clarify which convention we should use, harmonize and improve the format for doc-comments to achieve the envisioned goals.
+
+We use a tool called [nixdoc](https://github.com/nix-community/nixdoc) to use a `doc-comment'- like functionality, which allows us to build a subset of static documentation of the `nixpkgs.lib` functions.
+
+The format of a doc-comment was, however, never specified. The nix community does not understand the rules enforced by `nixdoc`, which we believe is one of the reasons for its lack of adoption.
+
+Here is an example of the format understood by *nixdoc*:
+
+```nix
+# nixpkgs/lib/attrsets.nix
+
+{
+  /* <Description Field>
+     Example:
+     <Comprehensive Code example>
+     Type:
+     <Type signature>
+  */
+  AttrFunc =
+    # <Desribe arg1>
+    arg1:
+    # <Describe arg2>
+    arg2:
+    
+    # ... implementation
+}
+```
+
+## Current problems
+
+### Lack of having an agreed Format
+
+The Nix community interprets the outer format and content of doc-comments differently. Within nixpkgs, several conventions and patterns still need to be harmonized. Some are compatible with nixdoc, while others do not and decide upon their conventions. We immediately observe one clear thing: Most of them utilize some form of markdown, even if the content is not rendered.
```

Comment:
```suggestion
The Nix community interprets the outer format and content of doc-comments differently. Within nixpkgs, several conventions and patterns still need to be harmonized. Some are compatible with nixdoc, while others are not use other styles. We immediately observe one clear thing: Most of them utilize some form of markdown, even if the content is not rendered.
```

I don't quite understand what you wanted to say here. Maybe this?

Reply:
I read original in diff as ", while others — do not, and instead opt for own conventions." or in other words "Some of them compatible with nixdoc, but not all.".

In suggested content ", while others are not use other styles." sounds like missing "and" to avoid reading "are not use". E.g.  ", while others — are not, and use other styles".